OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -42,12 +42,16 @@ public class FrozenActor
 		public bool NeedRenderables;
 		public bool IsRendering { get; private set; }
 
-		public FrozenActor(Actor self, MPos[] footprint, CellRegion footprintRegion, Shroud shroud)
+		public FrozenActor(Actor self, MPos[] footprint, Shroud shroud)
 		{
 			actor = self;
-			isVisibleTest = shroud.IsVisibleTest(footprintRegion);
+			isVisibleTest = shroud.IsVisibleTest;
```

Comment:
This is my own fault, but I've just realised this was an invalid optimization - you can't hold onto this delegate for so long. If somebody uses the dev setting to toggle shroud the results become incorrect since you're now applying the wrong delegate.

It needs to become a local variable in ```UpdateVisibility``` again.